### PR TITLE
fixed indexing for reporting singular errors

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**
These were off for some of the larger ones reported in pycycle. Luckily now we have _var_sizes cached in system so we don't have to try to sum up sizes anymore.

### CODE REVIEW

**Comment by Kenneth-T-Moore** [comment]
Yeah, wasn't sure I wanted to put that whole pycycle model in here.

**Comment by Kenneth-T-Moore** [comment]
Though actually, I've got an idea on how I can shrink this down to a smaller test.

**Comment by Kenneth-T-Moore** [comment]
Added tests, and good thing because there was an additional fix.

File: openmdao/solvers/linear/direct.py [hunk 1]
```diff
@@ -57,11 +57,11 @@ def format_singular_error(err, system, mtx):
...
     n = 0
     varname = "Unknown"
-    for name in system._var_allprocs_abs_names['output']:
-        relname = system._var_abs2prom['output'][name]
-        n += len(system._outputs[relname])
-        if loc <= n:
-            varname = relname
+    varsizes = system._var_sizes['nonlinear']['output']
+    for j, name in enumerate(system._var_allprocs_abs_names['output']):
+        n += varsizes[0][j]
```
Comment (naylor-b) [inline openmdao/solvers/linear/direct.py]: might want to use owning_rank here instead of 0.  Var might not exist on proc 0.

Comment (Kenneth-T-Moore) [reply openmdao/solvers/linear/direct.py]: Modified this to index via owning_rank.

**Pull request merged**